File: is_even.pl
% Prolog predicate to check if a number is even
% Usage: is_even(Number) succeeds if Number is even, fails otherwise

is_even(N) :-
    integer(N),           % Ensure N is an integer
    N mod 2 =:= 0.        % Check if N modulo 2 equals 0

% Alternative implementation using recursion for educational purposes
% is_even(0).
% is_even(N) :-
%     N > 0,
%     N2 is N - 2,
%     is_even(N2).
% is_even(N) :-
%     N < 0,
%     N2 is N + 2,
%     is_even(N2).

File: is_even.ml
let is_even n =
  n mod 2 = 0

let () =
  let test_num = 4 in
  if is_even test_num then
    Printf.printf "%d is even.\n" test_num
  else
    Printf.printf "%d is odd.\n" test_num;
  let test_num2 = 7 in
  if is_even test_num2 then
    Printf.printf "%d is even.\n" test_num2
  else
    Printf.printf "%d is odd.\n" test_num2